src/core/ScreenLockListenerPrivate.cpp

```diff
@@ -19,7 +19,7 @@
 #if defined(Q_OS_OSX)
 #include "ScreenLockListenerMac.h"
 #endif
-#if defined(Q_OS_LINUX)
+#if defined(Q_OS_UNIX)
```

Comment:
You need to exclude Q_OS_DARWIN here at least and I'm not sure about other BSDs. If in doubt, enable only Linux and OpenBSD (which you already did in CMakeLists.txt).